src/pages/Profile.tsx

```diff
@@ -653,6 +681,49 @@ class Profile extends Component<Props, ProfileState> {
           )}
         </div>
 
+        <div className="row">
+          <div className="col-md-12 text-center">
+            <button
+              onClick={() => {
+                this.loadUserSubscriptions(
+                  this.state.subscriptionsOffset + subscriptionsLimit
+                );
+              }}
+              hidden={
+                !this.state.hasMoreSubscriptions ||
+                (this.state.areSubscriptionsLoading &&
+                  this.state.hasMoreSubscriptions)
+              }
+              className="btn btn-secondary btn-lg btn-block"
+            >
+              {t("load_more_services")}
+            </button>
+            <div
+              className={
+                this.state.areSubscriptionsLoading
+                  ? "progress profile--services-progess"
+                  : "profile--services-progess-hidden"
+              }
+              style={
+                this.state.areSubscriptionsLoading
+                  ? { height: "56px" }
+                  : undefined
+              }
+            >
+              <div
+                className="progress-bar progress-bar-striped progress-bar-animated"
+                role="progressbar"
+                aria-valuenow={100}
+                aria-valuemin={0}
+                aria-valuemax={100}
+                style={{ width: "100%" }}
+              >
+                {t("loading_services")}
+              </div>
+            </div>
+          </div>
+        </div>
```

Comment:
Would it make sense to create a component? Locally in this file or in `components` folder for general availability, it's the same. The point is to have an easier-to-read render function (readers do not have to investigate all this structure to understand that it's a loader button)

Reply:
I Agree, let'so go with new dedicated `SubscriptionsLoader` component